packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -212,3 +216,175 @@ class _TweenAnimationBuilderState<T extends Object?>
     return widget.builder(context, _currentTween!.evaluate(animation), widget.child);
   }
 }
+
+/// Widget that animates an [Animatable] value and repeats indefinitely.
+///
+/// The animation continuously cycles from the [Animatable] evaluated at 0.0 to
+/// the value at 1.0.
+/// When [reverse] is true, the animation reverses direction when it reaches
+/// the end. When [paused] is true, the animation stops at its current value.
+///
+/// {@macro flutter.widgets.TweenAnimationBuilder.optimizations}
+///
+/// {@tool dartpad}
+/// This example shows a continuously rotating square that can be paused and resumed.
+///
+/// ** See code in examples/api/lib/widgets/tween_animation_builder/repeating_tween_animation_builder.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [TweenAnimationBuilder], which animates a tween value once.
+///  * [AnimationController.repeat], the underlying mechanism.
+class RepeatingTweenAnimationBuilder<T extends Object> extends StatefulWidget {
+  /// Creates a widget that repeats an animatable animation.
+  const RepeatingTweenAnimationBuilder({
+    super.key,
+    required this.animatable,
+    required this.duration,
+    this.curve = Curves.linear,
+    this.reverse = false,
+    this.paused = false,
+    required this.builder,
+    this.child,
+  });
+
+  /// The animatable to drive repeatedly.
+  ///
+  /// The animation continuously cycles from the value returned by evaluating
+  /// [animatable] at 0.0 to the value returned at 1.0.
+  /// In most cases this will be a [Tween] or [TweenSequence], but any
+  /// [Animatable] that produces values of type [T] is accepted.
+  ///
+  /// ## Ownership
```

Comment:
Now that it's an animatable this widget doesn't seem to mutate it anymore, and we should no longer discourage storing it. This instance should now be managed by the parent (which might not needs special mentioning any more).